lib/src/combined_wrappers/combined_list.dart

```diff
@@ -10,63 +10,66 @@ import 'dart:collection';
 /// concatenated list, but the underlying implementation is based on lazily
 /// accessing individual list instances.
 ///
-/// The returned list is unmodifiable.
+/// The resulting list has an index operator (`[]`) and `length` property that
+/// are both `O(lists)`, rather than `O(1)`, and the list is unmodifiable - but
+/// underlying changes to these lists are still accessible from the resulting
+/// list.
 List/*<T>*/ combineLists/*<T>*/(List<List/*<T>*/> lists) {
-  return new _CombinedList/*<T>*/(lists);
+  // Small optimization when there are no lists to avoid allocation.
+  if (lists.isEmpty) {
+    return const [];
```

Comment:
This still behaves inconsistently when `lists` itself is modified. I think we should just construct a `CombinedList()` regardless of the size of the input, and allow `lists` to change after the fact just like its contents.